lib/ClientFactory.php

```diff
@@ -9,10 +9,12 @@
   * @link      http://packagist.org/packages/sendgrid/sendgrid
   */
 
+namespace SendGrid;
+
 /**
   * Interface to the SendGrid Web API
   */
-class SendGrid
+class ClientFactory
```

Comment:
I'm going to write up (as a GitHub Issue) what I'm envisioning, similar to what [Adam did here](https://github.com/sendgrid/sendgrid-nodejs/pull/378). I hope you might find some time to me and helping flesh it out :) I'm sharing this context with you because I'm thinking what I propose is going to have an effect on what we ultimately decide to go with on this PR.